pyspecProcScripts/correlation_alignment.py

```diff
@@ -52,8 +61,20 @@ def correl_align(s, align_phases=False,tol=1e-4,indirect_dim='indirect',fig_titl
                 the width of the Gaussian function used to frequency filter
                 the data in the calculation of the correlation function.
     """
-    ph1_len = len(s.getaxis('ph1'))
-    ph2_len = len(s.getaxis('ph2'))
+    logging.info(strm("Applying the correlation routine"))
+    if nScans:
+        phcycdims = [j for j in s.dimlabels if j.startswith('ph')]
+        indirect = set(s.dimlabels)-set(phcycdims)-set([direct])
+        indirect = [j for j in s.dimlabels if j in indirect]
+        nScans_len = len(s.getaxis('nScans'))
```

Comment:
this will not hold up if the repeat/averaging dimension is called something else.  All you have to change, though, (I think) is to default it to `None`, and then replace any string instances of `'nScans'` with `nScans` --> making sure that they are always inside a conditional to check `if nScans:` (or `if nScans is not None:`, which should evaluate the same, I think)